In [199]:
import pandas as pd
import numpy as np
import json

from cleanup.cleanup_functions_collec import create_collec_staging
from cleanup.cleanup_siret_functions import clean_numeros

### Upload fichier

In [200]:
path = '../data/datasets/'

In [201]:
communities_df = pd.read_csv(path + 'all_communities_data.csv', delimiter=';', low_memory=False, index_col=False)

### Nettoyage de données

In [202]:
communities_clean = communities_df.copy()

In [203]:
communities_clean = create_collec_staging(communities_clean)

Cleaning duplicates: 36353 entries
After removing duplicates: 36353 entries
drop colummns: 18 columns
 After droping columns : 11 columns
Cleaning duplicates 'nom', 'type', and 'siren': 36353 entries
After removing duplicates: 36347 entries


In [204]:
communities_clean.shape

(36347, 11)

In [205]:
communities_clean[communities_clean['siren'].duplicated(keep=False)]

nom      siren type    cog code_dept  \
85     Métropole de Lyon  200046977  DEP    691       NaN   
91                 Paris  217500016  DEP     75       NaN   
29389              Paris  217500016  COM  75056        75   
35348  Métropole de Lyon  200046977  MET    NaN        69   

      code_departement_3digits code_region  population       epci  \
85                         NaN          84   1402326.0        NaN   
91                         NaN          11   2204773.0        NaN   
29389                      075        11.0   2204773.0  200054781   
35348                      069          84   1402326.0        NaN   

       trancheeffectifsunitelegale  effectifssup50  
85                            52.0            True  
91                            53.0            True  
29389                         53.0            True  
35348                         52.0            True

### Enrichir les données

In [206]:
communities_final = communities_clean.copy()

##### Ajout du code INSEE, qui servira de lien avec les autres sources de données en l'absence du SIREN.
* source georef-france-matching-siren-code.csv, https://public.opendatasoft.com/explore/dataset/georef-france-matching-siren-code/export/?disjunctive.abbr_type

In [207]:
siren_insee = pd.read_csv(path + "georef-france-matching-siren-code.csv", delimiter=';')

In [208]:
siren_insee.duplicated().sum()

0

In [209]:
siren_insee["code_insee"] = siren_insee["Code INSEE Officiel"].apply(clean_numeros)
siren_insee["siren"] = siren_insee["SIREN"].apply(clean_numeros)
siren_insee['type'] = siren_insee["Niveau"]

In [210]:
len('01001')

5

In [211]:
siren_insee[siren_insee["type"]=="COM"]["code_insee"].apply(lambda x: len(x)).value_counts()


code_insee
5    37475
Name: count, dtype: int64

In [212]:
siren_insee[siren_insee["type"]=="DEP"]["code_insee"].apply(lambda x: len(x)).value_counts()

code_insee
2    96
3     6
Name: count, dtype: int64

*Remarque:* Certaines sources n’ont pas le code INSEE des communes sur 5 chiffres. Il faut retirer le 0 devant ici.

In [213]:
mask = (siren_insee['type'] == 'COM') & (siren_insee['code_insee'].str.len() == 5)
siren_insee.loc[mask, 'code_insee'] = siren_insee.loc[mask, 'code_insee'].str.replace(r'^0', '', regex=True)

In [214]:
communities_final = communities_final.merge(siren_insee[['type', 'siren','code_insee']], on=('siren', 'type'), how='left')

In [215]:
communities_final[communities_final["code_insee"].isna()]["type"].value_counts()

type
CC     989
CA     227
MET     22
CU      14
EPT     11
CTU      3
DEP      1
Name: count, dtype: int64

#### 1. Latitude longitude
**source** : selected_communities_data.csv

In [216]:
selected_communities = pd.read_csv(path + 'selected_communities_data.csv', delimiter=';')

In [217]:
selected_communities.head()

Unnamed: 0                      nom      siren type cog cog_3digits  \
0           0               Guadeloupe  239710015  REG   1         NaN   
1           1            Île-de-France  237500079  REG  11         NaN   
2           2               Martinique  200055507  CTU   2         NaN   
3           3      Centre-Val de Loire  234500023  REG  24         NaN   
4           4  Bourgogne-Franche-Comté  200053726  REG  27         NaN   

  code_departement code_departement_3digits code_region  population  epci  \
0              NaN                      NaN         NaN    396153.0   NaN   
1              NaN                      NaN         NaN  12291279.0   NaN   
2              NaN                      NaN         NaN    377711.0   NaN   
3              NaN                      NaN         NaN   2634852.0   NaN   
4              NaN                      NaN         NaN   2885864.0   NaN   

                                             url_ptf  \
0  https://regionguadeloupe.opendatasoft.com/expl...   
1  https://data.iledefrance.fr/explore/?refine.pu...   
2                                                NaN   
3  https://data.centrevaldeloire.fr/explore/?sort...   
4  https://trouver.ternum-bfc.fr/organization/reg...   

                                        url_datagouv  \
0                                                NaN   
1  https://www.data.gouv.fr/fr/organizations/534f...   
2                                                NaN   
3  https://www.data.gouv.fr/fr/organizations/regi...   
4  https://www.data.gouv.fr/fr/organizations/5835...   

                id_datagouv         merge                                ptf  \
0                       NaN           ptf                           KaruData   
1  534fffa8a3a7292c64a780c8  ptf+datagouv            Ile-de-France, DataGouv   
2                       NaN           NaN                                NaN   
3  561a7f00c751df4e6dcdbb48  ptf+datagouv      Centre Val-de-Loire, DataGouv   
4  5835cdb988ee383e0bc65bb3  ptf+datagouv  Bourgogne-Franche-Comté, DataGouv   

   trancheeffectifsunitelegale  effectifssup50  longitude   latitude  
0                         42.0            True -61.568686  16.252882  
1                         53.0            True   2.504722  48.709167  
2                         51.0            True -61.015827  14.636792  
3                         51.0            True   1.685278  47.480556  
4                         51.0            True   4.809167  47.235278

In [218]:
selected_communities["siren"]= selected_communities["siren"].apply(clean_numeros)

In [219]:
communities_final = communities_final.merge(selected_communities[['siren', 'type', 'longitude','latitude']],on=['siren', 'type'], how='left' )

In [220]:
communities_final[communities_final["latitude"].isna()]['type'].value_counts()

type
COM    31842
CC         1
Name: count, dtype: int64

* **Sources** : 
    * 20230823-communes-departement-region.csv,https://www.data.gouv.fr/fr/datasets/r/dbe8a621-a9c4-4bc3-9cae-be1699c5ff25
    

In [221]:
com_dep_reg_df = pd.read_csv(path + '20230823-communes-departement-region.csv')

In [222]:
com_dep_reg_df.shape

(39201, 15)

In [223]:
com_dep_reg_df.head()

code_commune_INSEE       nom_commune_postal  code_postal  \
0               1001  L ABERGEMENT CLEMENCIAT         1400   
1               1002    L ABERGEMENT DE VAREY         1640   
2               1004        AMBERIEU EN BUGEY         1500   
3               1005      AMBERIEUX EN DOMBES         1330   
4               1006                  AMBLEON         1300   

      libelle_acheminement ligne_5   latitude  longitude  code_commune  \
0  L ABERGEMENT CLEMENCIAT     NaN  46.153426   4.926114           1.0   
1    L ABERGEMENT DE VAREY     NaN  46.009188   5.428017           2.0   
2        AMBERIEU EN BUGEY     NaN  45.960848   5.372926           4.0   
3      AMBERIEUX EN DOMBES     NaN  45.996180   4.912273           5.0   
4                  AMBLEON     NaN  45.749499   5.594320           6.0   

  article            nom_commune      nom_commune_complet code_departement  \
0      L'  Abergement-Clémenciat  L'Abergement-Clémenciat                1   
1      L'    Abergement-de-Varey    L'Abergement-de-Varey                1   
2     NaN      Ambérieu-en-Bugey        Ambérieu-en-Bugey                1   
3     NaN    Ambérieux-en-Dombes      Ambérieux-en-Dombes                1   
4     NaN                Ambléon                  Ambléon                1   

  nom_departement  code_region            nom_region  
0             Ain         84.0  Auvergne-Rhône-Alpes  
1             Ain         84.0  Auvergne-Rhône-Alpes  
2             Ain         84.0  Auvergne-Rhône-Alpes  
3             Ain         84.0  Auvergne-Rhône-Alpes  
4             Ain         84.0  Auvergne-Rhône-Alpes

In [224]:
com_dep_reg_df["code_insee"]=com_dep_reg_df['code_commune_INSEE']

In [225]:
com_dep_reg_df.isna().sum()

code_commune_INSEE          0
nom_commune_postal          0
code_postal                 0
libelle_acheminement        0
ligne_5                 35944
latitude                  269
longitude                 269
code_commune                7
article                 36621
nom_commune                 0
nom_commune_complet         0
code_departement            7
nom_departement           267
code_region               267
nom_region                267
code_insee                  0
dtype: int64

In [226]:

valeurs_dict_la = com_dep_reg_df.groupby('code_insee')['latitude'].first().to_dict()
valeurs_dict_lg = com_dep_reg_df.groupby('code_insee')['longitude'].first().to_dict()
communities_final['latitude'] = communities_final['latitude'].fillna(communities_final['code_insee'].map(valeurs_dict_la))
communities_final['longitude'] =communities_final['longitude'].fillna(communities_final['code_insee'].map(valeurs_dict_lg))


In [227]:
communities_final.isna().sum()

nom                               3
siren                             0
type                              1
cog                            1268
code_dept                       118
code_departement_3digits        118
code_region                      20
population                        2
epci                           1429
trancheeffectifsunitelegale    2109
effectifssup50                    0
code_insee                     1272
longitude                         5
latitude                          5
dtype: int64

In [228]:
communities_final.shape

(36352, 14)

In [229]:
# check vlue null of nom
communities_final[communities_final['siren'].duplicated(keep=False)]

nom      siren type    cog code_dept  \
85                 Métropole de Lyon  200046977  DEP    691       NaN   
91                             Paris  217500016  DEP     75       NaN   
10002                            NaN  200063782  COM  27676       NaN   
10003                            NaN  200063782  COM  27676       NaN   
29388                          Paris  217500016  COM  75056        75   
35347              Métropole de Lyon  200046977  MET    NaN        69   
35817  Métropole Européenne de Lille  200093201  MET    NaN        59   
35818  Métropole Européenne de Lille  200093201  MET    NaN        59   
35819  Métropole Européenne de Lille  200093201  MET    NaN        59   
35821        CA Agglomération d'Agen  200096956   CA    NaN        47   
35822        CA Agglomération d'Agen  200096956   CA    NaN        47   
36001             CA du Pays de l'Or  243400470   CA    NaN        34   
36002             CA du Pays de l'Or  243400470   CA    NaN        34   

      code_departement_3digits code_region  population       epci  \
85                         NaN          84   1402326.0        NaN   
91                         NaN          11   2204773.0        NaN   
10002                      NaN         NaN         NaN  200089456   
10003                      NaN         NaN         NaN  200089456   
29388                      075        11.0   2204773.0  200054781   
35347                      069          84   1402326.0        NaN   
35817                      059          32     24566.0        NaN   
35818                      059          32     24566.0        NaN   
35819                      059          32     24566.0        NaN   
35821                      047          75      5372.0        NaN   
35822                      047          75      5372.0        NaN   
36001                      034          76     45153.0        NaN   
36002                      034          76     45153.0        NaN   

       trancheeffectifsunitelegale  effectifssup50 code_insee  longitude  \
85                            52.0            True        691   4.832011   
91                            53.0            True        NaN   2.342222   
10002                         12.0           False      27676   1.295948   
10003                         12.0           False      27058   1.346657   
29388                         53.0            True      75056   2.347000   
35347                         52.0            True        NaN   4.835000   
35817                          NaN           False        NaN   3.045433   
35818                          NaN           False        NaN   3.045433   
35819                          NaN           False        NaN   3.045433   
35821                          NaN           False        NaN   0.631041   
35822                          NaN           False        NaN   0.631041   
36001                          NaN           False        NaN   3.976877   
36002                          NaN           False        NaN   3.976877   

        latitude  
85     45.757814  
91     48.856667  
10002  49.201220  
10003  49.235615  
29388  48.859000  
35347  45.758000  
35817  50.630992  
35818  50.630992  
35819  50.630992  
35821  44.202304  
35822  44.202304  
36001  43.587984  
36002  43.587984

*Remarque:* création de doublons

In [230]:
communities_final.drop_duplicates(inplace=True)

In [231]:
communities_final[communities_final['siren'].duplicated(keep=False)]

nom      siren type    cog code_dept  \
85     Métropole de Lyon  200046977  DEP    691       NaN   
91                 Paris  217500016  DEP     75       NaN   
10002                NaN  200063782  COM  27676       NaN   
10003                NaN  200063782  COM  27676       NaN   
29388              Paris  217500016  COM  75056        75   
35347  Métropole de Lyon  200046977  MET    NaN        69   

      code_departement_3digits code_region  population       epci  \
85                         NaN          84   1402326.0        NaN   
91                         NaN          11   2204773.0        NaN   
10002                      NaN         NaN         NaN  200089456   
10003                      NaN         NaN         NaN  200089456   
29388                      075        11.0   2204773.0  200054781   
35347                      069          84   1402326.0        NaN   

       trancheeffectifsunitelegale  effectifssup50 code_insee  longitude  \
85                            52.0            True        691   4.832011   
91                            53.0            True        NaN   2.342222   
10002                         12.0           False      27676   1.295948   
10003                         12.0           False      27058   1.346657   
29388                         53.0            True      75056   2.347000   
35347                         52.0            True        NaN   4.835000   

        latitude  
85     45.757814  
91     48.856667  
10002  49.201220  
10003  49.235615  
29388  48.859000  
35347  45.758000

*Remarque:* Une seule commune est dupliquée avec un nom nul et des coordonnées (latitude et longitude) différentes.

#### Densite et superficie 
**source**: communes-france-2025.csv

In [232]:
commune_france = pd.read_csv(path + 'communes-france-2025.csv',low_memory=False)

In [233]:
commune_france.columns

Index(['Unnamed: 0', 'code_insee', 'nom_standard', 'nom_sans_pronom', 'nom_a',
       'nom_de', 'nom_sans_accent', 'nom_standard_majuscule', 'typecom',
       'typecom_texte', 'reg_code', 'reg_nom', 'dep_code', 'dep_nom',
       'canton_code', 'canton_nom', 'epci_code', 'epci_nom', 'academie_code',
       'academie_nom', 'code_postal', 'codes_postaux', 'zone_emploi',
       'code_insee_centre_zone_emploi', 'code_unite_urbaine',
       'nom_unite_urbaine', 'taille_unite_urbaine',
       'type_commune_unite_urbaine', 'statut_commune_unite_urbaine',
       'population', 'superficie_hectare', 'superficie_km2', 'densite',
       'altitude_moyenne', 'altitude_minimale', 'altitude_maximale',
       'latitude_mairie', 'longitude_mairie', 'latitude_centre',
       'longitude_centre', 'grille_densite', 'grille_densite_texte',
       'niveau_equipements_services', 'niveau_equipements_services_texte',
       'gentile', 'url_wikipedia', 'url_villedereve'],
      dtype='object')

In [234]:
commune_france['typecom'].value_counts()

typecom
COM    34935
Name: count, dtype: int64

In [235]:
commune_france.head()

Unnamed: 0 code_insee             nom_standard        nom_sans_pronom  \
0           0      01001  L'Abergement-Clémenciat  Abergement-Clémenciat   
1           1      01002    L'Abergement-de-Varey    Abergement-de-Varey   
2           2      01004        Ambérieu-en-Bugey      Ambérieu-en-Bugey   
3           3      01005      Ambérieux-en-Dombes    Ambérieux-en-Dombes   
4           4      01006                  Ambléon                Ambléon   

                     nom_a                      nom_de  \
0  à Abergement-Clémenciat  de l'Abergement-Clémenciat   
1    à Abergement-de-Varey    de l'Abergement-de-Varey   
2      à Ambérieu-en-Bugey         d'Ambérieu-en-Bugey   
3    à Ambérieux-en-Dombes       d'Ambérieux-en-Dombes   
4                à Ambléon                   d'Ambléon   

           nom_sans_accent   nom_standard_majuscule typecom typecom_texte  \
0  l-abergement-clemenciat  L'ABERGEMENT-CLÉMENCIAT     COM       commune   
1    l-abergement-de-varey    L'ABERGEMENT-DE-VAREY     COM       commune   
2        amberieu-en-bugey        AMBÉRIEU-EN-BUGEY     COM       commune   
3      amberieux-en-dombes      AMBÉRIEUX-EN-DOMBES     COM       commune   
4                  ambleon                  AMBLÉON     COM       commune   

   ...  longitude_mairie latitude_centre longitude_centre grille_densite  \
0  ...             4.921          46.153            4.926              6   
1  ...             5.423          46.009            5.428              6   
2  ...             5.360          45.961            5.373              2   
3  ...             4.903          45.996            4.912              5   
4  ...             5.601          45.750            5.594              6   

             grille_densite_texte niveau_equipements_services  \
0        Rural à habitat dispersé                         0.0   
1        Rural à habitat dispersé                         0.0   
2  Centres urbains intermédiaires                         3.0   
3                   Bourgs ruraux                         1.0   
4        Rural à habitat dispersé                         0.0   

                   niveau_equipements_services_texte  \
0                                  communes non pôle   
1                                  communes non pôle   
2  centres structurants d'équipements et de services   
3        centres locaux d'équipements et de services   
4                                  communes non pôle   

                          gentile  \
0                             NaN   
1  Abergementais, Abergementaises   
2          Ambarrois, Ambarroises   
3                       Ambarrois   
4                      Ambléonais   

                                       url_wikipedia  \
0  https://fr.wikipedia.org/wiki/fr:L'Abergement-...   
1  https://fr.wikipedia.org/wiki/fr:L'Abergement-...   
2  https://fr.wikipedia.org/wiki/fr:Ambérieu-en-B...   
3  https://fr.wikipedia.org/wiki/fr:Ambérieux-en-...   
4           https://fr.wikipedia.org/wiki/fr:Ambléon   

                                     url_villedereve  
0  https://villedereve.fr/ville/01001-l-abergemen...  
1  https://villedereve.fr/ville/01002-l-abergemen...  
2  https://villedereve.fr/ville/01004-amberieu-en...  
3  https://villedereve.fr/ville/01005-amberieux-e...  
4         https://villedereve.fr/ville/01006-ambleon  

[5 rows x 47 columns]

In [236]:
commune_france.isna().sum()

Unnamed: 0                               0
code_insee                               0
nom_standard                             0
nom_sans_pronom                          0
nom_a                                    0
nom_de                                   0
nom_sans_accent                          0
nom_standard_majuscule                   0
typecom                                  0
typecom_texte                            0
reg_code                                 0
reg_nom                                  0
dep_code                                 0
dep_nom                                  0
canton_code                             56
canton_nom                              56
epci_code                                3
epci_nom                                 3
academie_code                            0
academie_nom                             0
code_postal                              3
codes_postaux                            3
zone_emploi                              3
code_insee_

In [237]:
commune_france['type']=commune_france['typecom']


In [238]:
mask = (commune_france['type'] == 'COM') & (commune_france['code_insee'].str.len() == 5)
commune_france.loc[mask, 'code_insee'] = commune_france.loc[mask, 'code_insee'].str.replace(r'^0', '', regex=True)

In [239]:
communities_final = communities_final.merge(commune_france[['code_insee', 'type', 'superficie_km2','densite']], on=['code_insee', 'type'], how='left')

In [240]:
communities_final.isna().sum()

nom                               3
siren                             0
type                              1
cog                            1264
code_dept                       118
code_departement_3digits        118
code_region                      20
population                        2
epci                           1425
trancheeffectifsunitelegale    2105
effectifssup50                    0
code_insee                     1268
longitude                         5
latitude                          5
superficie_km2                 1416
densite                        1416
dtype: int64

In [241]:
communities_final.head()

nom      siren type cog code_dept  \
0               Guadeloupe  239710015  REG   1       NaN   
1            Île-de-France  237500079  REG  11       NaN   
2               Martinique  200055507  CTU   2       NaN   
3      Centre-Val de Loire  234500023  REG  24       NaN   
4  Bourgogne-Franche-Comté  200053726  REG  27       NaN   

  code_departement_3digits code_region  population epci  \
0                      NaN         NaN    396153.0  NaN   
1                      NaN         NaN  12291279.0  NaN   
2                      NaN         NaN    377711.0  NaN   
3                      NaN         NaN   2634852.0  NaN   
4                      NaN         NaN   2885864.0  NaN   

   trancheeffectifsunitelegale  effectifssup50 code_insee  longitude  \
0                         42.0            True         01 -61.568686   
1                         53.0            True         11   2.504722   
2                         51.0            True        NaN -61.015827   
3                         51.0            True         24   1.685278   
4                         51.0            True         27   4.809167   

    latitude  superficie_km2  densite  
0  16.252882             NaN      NaN  
1  48.709167             NaN      NaN  
2  14.636792             NaN      NaN  
3  47.480556             NaN      NaN  
4  47.235278             NaN      NaN

#### Compléter les codes départements pour faire le lien  avec les communes ou autres collectivités
source nocodb :ofgl-base-departements-consolidee.csv

In [242]:
bases_departements = pd.read_csv(path +'ofgl-base-departements-consolidee.csv', delimiter=';')

In [243]:
bases_departements['siren'] = bases_departements['siren'].apply(clean_numeros)
bases_departements['dep_code'] = bases_departements['dep_code'].apply(lambda x: str(x).strip().upper() if pd.notna(x) else x)

In [244]:
valeurs_dict_dep = bases_departements.groupby('siren')['dep_code'].first().to_dict()
communities_final['code_dept'] = communities_final['code_dept'].fillna(communities_final['siren'].map(valeurs_dict_dep))

In [245]:
communities_final[communities_final["code_dept"].isnull()]["type"].value_counts()

type
REG    14
CTU     3
COM     2
Name: count, dtype: int64

In [246]:
communities_final[(communities_final["code_dept"].isnull()) & (communities_final["type"]=='COM')]

nom      siren type    cog code_dept code_departement_3digits  \
10002  NaN  200063782  COM  27676       NaN                      NaN   
10003  NaN  200063782  COM  27676       NaN                      NaN   

      code_region  population       epci  trancheeffectifsunitelegale  \
10002         NaN         NaN  200089456                         12.0   
10003         NaN         NaN  200089456                         12.0   

       effectifssup50 code_insee  longitude   latitude  superficie_km2  \
10002           False      27676   1.295948  49.201220            15.0   
10003           False      27058   1.346657  49.235615             NaN   

       densite  
10002    117.0  
10003      NaN

##### Remarque: une seule commune avec nom null et code dept null: siren: 200063782

#### Compléter les codes régions pour faire le lien  avec les communes, départements ou autres collectivités
Sources: NocoDB ofgl-base-regions-consolidee.csv 

In [247]:
communities_final[communities_final['code_region'].isna()]["type"].value_counts()

type
REG    14
CTU     3
COM     2
Name: count, dtype: int64

In [248]:
base_regions = pd.read_csv(path + 'ofgl-base-regions-consolidee.csv', delimiter=';')

In [249]:
base_regions['siren'] = base_regions['siren'].apply(clean_numeros)
base_regions['reg_code'] = base_regions['reg_code'].apply(lambda x: str(x).strip().upper() if pd.notna(x) else x)

In [250]:
valeurs_dict_reg = base_regions.groupby('siren')['reg_code'].first().to_dict()
communities_final['code_region'] = communities_final['code_region'].fillna(communities_final['siren'].map(valeurs_dict_reg))

In [251]:
communities_final[communities_final['code_region'].isna()]["type"].value_counts()

type
COM    2
Name: count, dtype: int64

Remarque: la même commune avec code région null: 200063782

#### Les élus

##### Ajout elus minicipaux
* source: nocodb: elus-conseillers-municipaux-cm.csv

In [252]:
elus_minicip = pd.read_csv(path + "elus-conseillers-municipaux-cm.csv", delimiter=';', low_memory=False)

In [253]:
elus_minicip.shape

(488719, 16)

In [254]:
elus_minicip.columns

Index(['Code du département', 'Libellé du département',
       'Code de la collectivité à statut particulier',
       'Libellé de la collectivité à statut particulier', 'Code de la commune',
       'Libellé de la commune', 'Nom de l'élu', 'Prénom de l'élu', 'Code sexe',
       'Date de naissance', 'Code de la catégorie socio-professionnelle',
       'Libellé de la catégorie socio-professionnelle',
       'Date de début du mandat', 'Libellé de la fonction',
       'Date de début de la fonction', 'Code nationalité'],
      dtype='object')

In [255]:
elus_minicip.isna().sum()

Code du département                                  3315
Libellé du département                               3315
Code de la collectivité à statut particulier       481236
Libellé de la collectivité à statut particulier    481236
Code de la commune                                      0
Libellé de la commune                                   0
Nom de l'élu                                            1
Prénom de l'élu                                         0
Code sexe                                               0
Date de naissance                                       0
Code de la catégorie socio-professionnelle            142
Libellé de la catégorie socio-professionnelle           0
Date de début du mandat                                 0
Libellé de la fonction                             338034
Date de début de la fonction                       338034
Code nationalité                                        0
dtype: int64

In [256]:
elus_minicip["Libellé de la fonction"].value_counts()

Libellé de la fonction
Maire                     34844
1er adjoint au Maire      33863
2ème adjoint au Maire     31238
3ème adjoint au Maire     19969
4ème adjoint au Maire     11000
5ème adjoint au Maire      5839
6ème adjoint au Maire      3820
7ème adjoint au Maire      2826
Maire délégué              2286
8ème adjoint au Maire      2215
9ème adjoint au Maire       917
10ème adjoint au Maire      466
11ème adjoint au Maire      326
12ème adjoint au Maire      246
13ème adjoint au Maire      193
14ème adjoint au Maire      148
15ème adjoint au Maire      104
16ème adjoint au Maire       92
17ème adjoint au Maire       71
18ème adjoint au Maire       57
19ème adjoint au Maire       42
20ème adjoint au Maire       37
21ème adjoint au Maire       29
22ème adjoint au Maire       12
25ème adjoint au Maire        8
24ème adjoint au Maire        8
23ème adjoint au Maire        8
26ème adjoint au Maire        5
29ème adjoint au Maire        2
37ème adjoint au Maire        2
36ème adjoint au 

In [257]:
elus_minicip_maire = elus_minicip[elus_minicip["Libellé de la fonction"] == "Maire"]

In [258]:
elus_minicip_maire["code_insee"] = elus_minicip_maire["Code de la commune"]
elus_minicip_maire['type_elu'] = "Maire"
elus_minicip_maire['nom_elu'] = elus_minicip_maire["Nom de l'élu"]
elus_minicip_maire['type'] = "COM"

/tmp/ipykernel_62332/2562459737.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  elus_minicip_maire["code_insee"] = elus_minicip_maire["Code de la commune"]
/tmp/ipykernel_62332/2562459737.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  elus_minicip_maire['type_elu'] = "Maire"
/tmp/ipykernel_62332/2562459737.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: ht

In [259]:
elus_minicip_maire.head()

Code du département Libellé du département  \
3                    1                    Ain   
21                   1                    Ain   
37                   1                    Ain   
60                   1                    Ain   
77                   1                    Ain   

    Code de la collectivité à statut particulier  \
3                                            NaN   
21                                           NaN   
37                                           NaN   
60                                           NaN   
77                                           NaN   

   Libellé de la collectivité à statut particulier Code de la commune  \
3                                              NaN               1001   
21                                             NaN               1002   
37                                             NaN               1004   
60                                             NaN               1005   
77                                             NaN               1006   

      Libellé de la commune Nom de l'élu Prénom de l'élu Code sexe  \
3   L'Abergement-Clémenciat       BOULON          Daniel         M   
21    L'Abergement-De-Varey        ORSET             Max         M   
37        Ambérieu-En-Bugey        FABRE          Daniel         M   
60      Ambérieux-En-Dombes       FORNES       Christine         F   
77                  Ambléon       BIONDA           Annie         F   

   Date de naissance  Code de la catégorie socio-professionnelle  \
3         04/03/1951                                        74.0   
21        02/11/1947                                        65.0   
37        07/09/1961                                        74.0   
60        06/07/1962                                        21.0   
77        28/11/1951                                        77.0   

        Libellé de la catégorie socio-professionnelle Date de début du mandat  \
3                                        Ancien cadre              18/05/2020   
21  Ouvrier qualifié de la manutention, du magasin...              18/05/2020   
37                                       Ancien cadre              18/05/2020   
60                                            Artisan              02/05/2021   
77                                     Ancien employé              18/05/2020   

   Libellé de la fonction Date de début de la fonction Code nationalité  \
3                   Maire                   26/05/2020               FR   
21                  Maire                   27/05/2020               FR   
37                  Maire                   28/05/2020               FR   
60                  Maire                   07/05/2021               FR   
77                  Maire                   25/05/2020               FR   

   code_insee type_elu nom_elu type  
3        1001    Maire  BOULON  COM  
21       1002    Maire   ORSET  COM  
37       1004    Maire   FABRE  COM  
60       1005    Maire  FORNES  COM  
77       1006    Maire  BIONDA  COM

In [260]:
communities_final = communities_final.merge(elus_minicip_maire[['code_insee','nom_elu','type_elu', 'type']], on=('code_insee','type'), how='left')

In [261]:

communities_final[communities_final['type_elu'].notna()]['type'].value_counts()

type
COM    34758
Name: count, dtype: int64

##### Ajout elus departementaux
* source: nocodb: elus-conseillers-departementaux-cd.csv

In [262]:
elus_dept = pd.read_csv(path + "elus-conseillers-departementaux-cd.csv", delimiter=';')


In [263]:
elus_dept.head()

Code du département Libellé du département  Code du canton  \
0                  01                    Ain             101   
1                  01                    Ain             101   
2                  01                    Ain             102   
3                  01                    Ain             102   
4                  01                    Ain             103   

   Libellé du canton Nom de l'élu Prénom de l'élu Code sexe Date de naissance  \
0  Ambérieu-En-Bugey       BRUNET            Joël         M        02/05/1955   
1  Ambérieu-En-Bugey        PETIT         Aurélie         F        29/08/1982   
2           Attignat     FOURNIER        Clotilde         F        01/09/1966   
3           Attignat       MARTIN          Walter         M        07/05/1966   
4         Valserhône    LARMANJAT             Guy         M        28/09/1962   

   Code de la catégorie socio-professionnelle  \
0                                          23   
1                                          38   
2                                          35   
3                                          38   
4                                          38   

       Libellé de la catégorie socio-professionnelle Date de début du mandat  \
0           Chef d'entreprise de 10 salariés ou plus              01/07/2021   
1          Ingénieur et cadre technique d'entreprise              01/07/2021   
2  Profession de l'information, des arts et des s...              01/07/2021   
3          Ingénieur et cadre technique d'entreprise              01/07/2021   
4          Ingénieur et cadre technique d'entreprise              01/07/2021   

                         Libellé de la fonction Date de début de la fonction  
0                                           NaN                          NaN  
1                                           NaN                          NaN  
2  9ème Vice-président du conseil départemental                   01/07/2021  
3                                           NaN                          NaN  
4                                           NaN                          NaN

In [264]:
elus_dept["Libellé de la fonction"].value_counts()

Libellé de la fonction
Président du conseil départemental                  95
4ème Vice-président du conseil départemental        94
6ème Vice-président du conseil départemental        94
2ème Vice-président du conseil départemental        93
1er Vice-président du conseil départemental         92
7ème Vice-président du conseil départemental        92
3ème Vice-président du conseil départemental        91
5ème Vice-président du conseil départemental        91
8ème Vice-président du conseil départemental        87
9ème Vice-président du conseil départemental        80
10ème Vice-président du conseil départemental       71
11ème Vice-président du conseil départemental       54
12ème Vice-président du conseil départemental       46
13ème Vice-président du conseil départemental       28
14ème Vice-président du conseil départemental       20
15ème Vice-président du conseil départemental       19
Président de commission du conseil départemental    17
Name: count, dtype: int64

In [265]:
elus_dept_president = elus_dept[elus_dept["Libellé de la fonction"] == "Président du conseil départemental"]

In [266]:
elus_dept_president['type_elu'] = "Président du conseil départemental"
elus_dept_president['nom_elu'] = elus_dept["Nom de l'élu"]
elus_dept_president["type"] = "DEP"
elus_dept_president["code_dept"] =  elus_dept_president["Code du département"]

/tmp/ipykernel_62332/3650462116.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  elus_dept_president['type_elu'] = "Président du conseil départemental"
/tmp/ipykernel_62332/3650462116.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  elus_dept_president['nom_elu'] = elus_dept["Nom de l'élu"]
/tmp/ipykernel_62332/3650462116.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the docu

In [267]:
valeurs_dict_nomelu_dept = elus_dept_president.groupby('code_dept')['nom_elu'].first().to_dict()
communities_final.loc[communities_final['type'] == 'DEP', 'nom_elu'] = (
    communities_final.loc[communities_final['type'] == 'DEP', 'nom_elu']
    .fillna(communities_final['code_dept'].map(valeurs_dict_nomelu_dept))
)
communities_final.loc[communities_final['type'] == 'DEP', 'type_elu'] = (
    communities_final.loc[communities_final['type'] == 'DEP', 'type_elu']
    .fillna("Président du conseil départemental")
)

In [268]:
communities_final[communities_final['type']=='DEP']

nom      siren type  cog code_dept  \
17                       Ain  220100010  DEP   01        01   
18                     Aisne  220200026  DEP   02        02   
19                    Allier  220300016  DEP   03        03   
20   Alpes-de-Haute-Provence  220400014  DEP   04        04   
21              Hautes-Alpes  220500011  DEP   05        05   
..                       ...        ...  ...  ...       ...   
110             Val-de-Marne  229400288  DEP   94        94   
111               Val-d'Oise  229501275  DEP   95        95   
112               Guadeloupe  229710017  DEP  971       971   
113               La Réunion  229740014  DEP  974       974   
114                  Mayotte  229850003  DEP  976       976   

    code_departement_3digits code_region  population epci  \
17                       NaN          84    659180.0  NaN   
18                       NaN          32    546527.0  NaN   
19                       NaN          84    347035.0  NaN   
20                       NaN          93    168381.0  NaN   
21                       NaN          93    145883.0  NaN   
..                       ...         ...         ...  ...   
110                      NaN          11   1397035.0  NaN   
111                      NaN          11   1239262.0  NaN   
112                      NaN           1    396153.0  NaN   
113                      NaN           4    863063.0  NaN   
114                      NaN           6    262895.0  NaN   

     trancheeffectifsunitelegale  effectifssup50 code_insee  longitude  \
17                          51.0            True         01   5.348889   
18                          51.0            True         02   3.558333   
19                          42.0            True         03   3.188333   
20                          42.0            True         04   6.243889   
21                          42.0            True         05   6.263056   
..                           ...             ...        ...        ...   
110                         52.0            True         94   2.468889   
111                         51.0            True         95   2.131111   
112                         51.0            True        971 -61.568686   
113                         52.0            True        974 -21.114533   
114                          NaN           False        976 -52.999998   

      latitude  superficie_km2  densite     nom_elu  \
17   46.099444             NaN      NaN    DEGUERRY   
18   49.559444             NaN      NaN  FRICOTEAUX   
19   46.393611             NaN      NaN    RIBOULET   
20   44.106111             NaN      NaN   BARREILLE   
21   44.663611             NaN      NaN     BERNARD   
..         ...             ...      ...         ...   
110  48.777500             NaN      NaN   CAPITANIO   
111  49.082778             NaN      NaN    CAVECCHI   
112  16.252882             NaN      NaN      LOSBAR   
113  55.532062             NaN      NaN    MELCHIOR   
114   4.003989             NaN      NaN     OUSSENI   

                               type_elu  
17   Président du conseil départemental  
18   Président du conseil départemental  
19   Président du conseil départemental  
20   Président du conseil départemental  
21   Président du conseil départemental  
..                                  ...  
110  Président du conseil départemental  
111  Président du conseil départemental  
112  Président du conseil départemental  
113  Président du conseil départemental  
114  Président du conseil départemental  

[98 rows x 18 columns]

##### Ajout elus regionaux
* Source: nocodb: elus-conseillers-regionaux-cr.csv

In [269]:
elus_reg = pd.read_csv(path + 'elus-conseillers-regionaux-cr.csv', delimiter=';')
elus_reg

Code de la région        Libellé de la région  \
0                     1                  Guadeloupe   
1                     1                  Guadeloupe   
2                     1                  Guadeloupe   
3                     1                  Guadeloupe   
4                     1                  Guadeloupe   
...                 ...                         ...   
1744                 93  Provence-Alpes-Cote D'Azur   
1745                 93  Provence-Alpes-Cote D'Azur   
1746                 93  Provence-Alpes-Cote D'Azur   
1747                 93  Provence-Alpes-Cote D'Azur   
1748                 93  Provence-Alpes-Cote D'Azur   

     Code de la section départementale Libellé de la section départementale  \
0                                  971                           Guadeloupe   
1                                  971                           Guadeloupe   
2                                  971                           Guadeloupe   
3                                  971                           Guadeloupe   
4                                  971                           Guadeloupe   
...                                ...                                  ...   
1744                                84                             Vaucluse   
1745                                84                             Vaucluse   
1746                                84                             Vaucluse   
1747                                84                             Vaucluse   
1748                                84                             Vaucluse   

      Nom de l'élu  Prénom de l'élu Code sexe Date de naissance  \
0         ARMOUGOM  Betty Véronique         F        09/07/1965   
1          BAILLET         Patricia         F        17/06/1970   
2          BARDAIL             Jean         M        30/05/1951   
3        BITUFWILA          Aurélie         F        21/08/1991   
4     BONDOT-GALAS         Gersiane         F        09/02/1963   
...            ...              ...       ...               ...   
1744      PERILHOU    Jean-François         M        05/09/1982   
1745       RIGAULT      Anne-Sophie         F        23/03/1976   
1746       RIMBERT        Catherine         F        13/09/1968   
1747        RIPERT           Gilles         M        04/06/1958   
1748         VIALA           Sylvie         F        07/07/1954   

      Code de la catégorie socio-professionnelle  \
0                                           23.0   
1                                           37.0   
2                                           75.0   
3                                           42.0   
4                                           42.0   
...                                          ...   
1744                                        31.0   
1745                                        85.0   
1746                                        31.0   
1747                                        31.0   
1748                                        34.0   

          Libellé de la catégorie socio-professionnelle  \
0              Chef d'entreprise de 10 salariés ou plus   
1        Cadre administratif et commercial d'entreprise   
2                     Ancienne profession intermédiaire   
3        Professeur des écoles, instituteur et assimilé   
4        Professeur des écoles, instituteur et assimilé   
...                                                 ...   
1744                                Profession libérale   
1745  Personne diverse sans activité professionnelle...   
1746                                Profession libérale   
1747                                Profession libérale   
1748                Professeur, profession scientifique   

     Date de début du mandat                   Libellé de la fonction  \
0                 02/07/2021                                      NaN   
1                 02/07/2021                                      NaN   
2                 02/07/2021  3ème Vic

In [270]:
communities_final[communities_final['type']=='DEP']['code_region']

17     84
18     32
19     84
20     93
21     93
       ..
110    11
111    11
112     1
113     4
114     6
Name: code_region, Length: 98, dtype: object

In [271]:
elus_reg["Libellé de la fonction"].value_counts()

Libellé de la fonction
3ème Vice-président du conseil régional     14
8ème Vice-président du conseil régional     14
Président du conseil régional               14
4ème Vice-président du conseil régional     14
2ème Vice-président du conseil régional     14
1er Vice-président du conseil régional      13
6ème Vice-président du conseil régional     13
9ème Vice-président du conseil régional     13
7ème Vice-président du conseil régional     12
5ème Vice-président du conseil régional     12
10ème Vice-président du conseil régional    12
12ème Vice-président du conseil régional    12
11ème Vice-président du conseil régional    12
13ème Vice-président du conseil régional    12
15ème Vice-président du conseil régional    10
14ème Vice-président du conseil régional     9
Name: count, dtype: int64

In [272]:
elus_reg_president = elus_reg[elus_reg["Libellé de la fonction"]=="Président du conseil régional"]

In [273]:
elus_reg_president['type_elu'] = "Président du conseil régional"
elus_reg_president['nom_elu'] = elus_reg["Nom de l'élu"]
elus_reg_president["type"] = "REG"
elus_reg_president["code_region"] =  elus_reg_president["Code de la région"].apply(lambda x: str(x).strip() if pd.notna(x) else x)

/tmp/ipykernel_62332/2490365004.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  elus_reg_president['type_elu'] = "Président du conseil régional"
/tmp/ipykernel_62332/2490365004.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  elus_reg_president['nom_elu'] = elus_reg["Nom de l'élu"]
/tmp/ipykernel_62332/2490365004.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentatio

In [274]:
valeurs_dict_nomelu_reg = elus_reg_president.groupby('code_region')['nom_elu'].first().to_dict()

# Remplacer les valeurs NaN dans 'nom_elu' pour les régions
communities_final.loc[communities_final['type'] == 'REG', 'nom_elu'] = (
    communities_final.loc[communities_final['type'] == 'REG', 'nom_elu']
    .fillna(communities_final['code_region'].map(valeurs_dict_nomelu_reg))
)
communities_final.loc[communities_final['type'] == 'REG', 'type_elu'] = (
    communities_final.loc[communities_final['type'] == 'REG', 'type_elu']
    .fillna("Président du conseil régional")
)

In [275]:
valeurs_dict_nomelu_reg

{'1': 'CHALUS',
 '11': 'PÉCRESSE',
 '24': 'BONNEAU',
 '27': 'DUFAY',
 '28': 'MORIN',
 '32': 'BERTRAND',
 '4': 'BELLO',
 '44': 'LEROY',
 '52': 'MORANÇAIS',
 '53': 'CHESNAIS-GIRARD',
 '75': 'ROUSSET',
 '76': 'DELGA',
 '84': 'PANNEKOUCKE',
 '93': 'MUSELIER'}

In [276]:
communities_final[communities_final["type"]=="REG"]

nom      siren type cog code_dept  \
0                   Guadeloupe  239710015  REG   1       NaN   
1                Île-de-France  237500079  REG  11       NaN   
3          Centre-Val de Loire  234500023  REG  24       NaN   
4      Bourgogne-Franche-Comté  200053726  REG  27       NaN   
5                    Normandie  200053403  REG  28       NaN   
7              Hauts-de-France  200053742  REG  32       NaN   
8                   La Réunion  239740012  REG   4       NaN   
9                    Grand Est  200052264  REG  44       NaN   
10            Pays de la Loire  234400034  REG  52       NaN   
11                    Bretagne  233500016  REG  53       NaN   
12          Nouvelle-Aquitaine  200053759  REG  75       NaN   
13                   Occitanie  200053791  REG  76       NaN   
14        Auvergne-Rhône-Alpes  200053767  REG  84       NaN   
15  Provence-Alpes-Côte d'Azur  231300021  REG  93       NaN   

   code_departement_3digits code_region  population epci  \
0                       NaN           1    396153.0  NaN   
1                       NaN          11  12291279.0  NaN   
3                       NaN          24   2634852.0  NaN   
4                       NaN          27   2885864.0  NaN   
5                       NaN          28   3403309.0  NaN   
7                       NaN          32   6096177.0  NaN   
8                       NaN           4    863063.0  NaN   
9                       NaN          44   5657093.0  NaN   
10                      NaN          52   3846161.0  NaN   
11                      NaN          53   3407564.0  NaN   
12                      NaN          75   6094367.0  NaN   
13                      NaN          76   5968795.0  NaN   
14                      NaN          84   8120668.0  NaN   
15                      NaN          93   5106679.0  NaN   

    trancheeffectifsunitelegale  effectifssup50 code_insee  longitude  \
0                          42.0            True         01 -61.568686   
1                          53.0            True         11   2.504722   
3                          51.0            True         24   1.685278   
4                          51.0            True         27   4.809167   
5                          52.0            True         28   0.106667   
7                          52.0            True         32   2.775278   
8                          51.0            True         04 -21.114533   
9                          52.0            True         44   5.619444   
10                         51.0            True         52   0.823889   
11                         51.0            True         53  -1.161389   
12                         52.0            True         75   0.197778   
13                         52.0            True         76   2.137222   
14                         52.0            True         84   4.538056   
15                         52.0            True         93   6.053333   

     latitude  superficie_km2  densite          nom_elu  \
0   16.252882             NaN      NaN           CHALUS   
1   48.709167             NaN      NaN         PÉCRESSE   
3   47.480556             NaN      NaN          BONNEAU   
4   47.235278             NaN      NaN            DUFAY   
5   49.121111             NaN      NaN            MORIN   
7   49.966111             NaN      NaN         BERTRAND   
8   55.532062             NaN      NaN            BELLO   
9   48.689167             NaN      NaN            LEROY   
10  47.474722             NaN      NaN        MORANÇAIS   
11  48.179722             NaN      NaN  CHESNAIS-GIRARD   
12  45.192222             NaN      NaN          ROUSSET   
13  43.702222             NaN      NaN            DELGA   
14  45.515833             NaN      NaN      PANNEKOUCKE   
15  43.955000             NaN      NaN         MUSELIER   

                         type_elu  
0   Président du conseil régional  
1   Président du conseil régional  
3   Président du conseil régional  
4   Président du conseil régional  
5  

#### Obligation publication

In [277]:
## TODO

In [278]:
communities_final[communities_final['type']=='COM']

nom      siren type    cog code_dept  \
115    L'Abergement-Clémenciat  210100012  COM  01001        01   
116      L'Abergement-de-Varey  210100020  COM  01002        01   
117          Ambérieu-en-Bugey  210100046  COM  01004        01   
118        Ambérieux-en-Dombes  210100053  COM  01005        01   
119                    Ambléon  210100061  COM  01006        01   
...                        ...        ...  ...    ...       ...   
35079            M'Tsangamouji  200008829  COM  97613       976   
35080                 Ouangani  200008852  COM  97614       976   
35081                 Pamandzi  200008860  COM  97615       976   
35082                     Sada  200008878  COM  97616       976   
35083                 Tsingoni  200008886  COM  97617       976   

      code_departement_3digits code_region  population       epci  \
115                        001        84.0       794.0  200069193   
116                        001        84.0       249.0  240100883   
117                        001        84.0     14428.0  240100883   
118                        001        84.0      1723.0  200042497   
119                        001        84.0       117.0  200040350   
...                        ...         ...         ...        ...   
35079                      976         6.0      6586.0  200059871   
35080                      976         6.0     10393.0  200059871   
35081                      976         6.0     11802.0  200050532   
35082                      976         6.0     11619.0  200059871   
35083                      976         6.0     14235.0  200059871   

       trancheeffectifsunitelegale  effectifssup50 code_insee  longitude  \
115                           11.0           False       1001   4.926114   
116                            2.0           False       1002   5.428017   
117                           32.0            True       1004   5.365228   
118                           12.0           False       1005   4.912273   
119                            1.0           False       1006   5.594320   
...                            ...             ...        ...        ...   
35079                          NaN           False      97613  45.087170   
35080                          NaN           False      97614  45.137910   
35081                          NaN           False      97615  45.284206   
35082                          NaN           False      97616  45.118550   
35083                          NaN           False      97617  45.134428   

        latitude  superficie_km2  densite    nom_elu type_elu  
115    46.153426            16.0     53.0     BOULON    Maire  
116    46.009188             9.0     29.0      ORSET    Maire  
117    45.959530            24.0    607.0      FABRE    Maire  
118    45.996180            16.0    118.0     FORNES    Maire  
119    45.749499             6.0     19.0     BIONDA    Maire  
...          ...             ...      ...        ...      ...  
35079 -12.751310            22.0    298.0   IBRAHIMA    Maire  
35080 -12.837096            18.0    558.0      AMBDI    Maire  
35081 -12.796135             4.0   2686.0  MADI SOUF    Maire  
35082 -12.861165            11.0   1028.0   ABDALLAH    Maire  
35083 -12.782167            34.0    407.0  ISSILAMOU    Maire  

[34969 rows x 18 columns]

In [279]:
## remetre le 0 pour le code_insee des communes
mask = (communities_final['type'] == 'COM') & (communities_final['code_insee'].str.len() < 5)
communities_final.loc[mask, 'code_insee'] = communities_final.loc[mask, 'code_insee'].str.zfill(5)

In [280]:
communities_final[communities_final['type']=='COM']

nom      siren type    cog code_dept  \
115    L'Abergement-Clémenciat  210100012  COM  01001        01   
116      L'Abergement-de-Varey  210100020  COM  01002        01   
117          Ambérieu-en-Bugey  210100046  COM  01004        01   
118        Ambérieux-en-Dombes  210100053  COM  01005        01   
119                    Ambléon  210100061  COM  01006        01   
...                        ...        ...  ...    ...       ...   
35079            M'Tsangamouji  200008829  COM  97613       976   
35080                 Ouangani  200008852  COM  97614       976   
35081                 Pamandzi  200008860  COM  97615       976   
35082                     Sada  200008878  COM  97616       976   
35083                 Tsingoni  200008886  COM  97617       976   

      code_departement_3digits code_region  population       epci  \
115                        001        84.0       794.0  200069193   
116                        001        84.0       249.0  240100883   
117                        001        84.0     14428.0  240100883   
118                        001        84.0      1723.0  200042497   
119                        001        84.0       117.0  200040350   
...                        ...         ...         ...        ...   
35079                      976         6.0      6586.0  200059871   
35080                      976         6.0     10393.0  200059871   
35081                      976         6.0     11802.0  200050532   
35082                      976         6.0     11619.0  200059871   
35083                      976         6.0     14235.0  200059871   

       trancheeffectifsunitelegale  effectifssup50 code_insee  longitude  \
115                           11.0           False      01001   4.926114   
116                            2.0           False      01002   5.428017   
117                           32.0            True      01004   5.365228   
118                           12.0           False      01005   4.912273   
119                            1.0           False      01006   5.594320   
...                            ...             ...        ...        ...   
35079                          NaN           False      97613  45.087170   
35080                          NaN           False      97614  45.137910   
35081                          NaN           False      97615  45.284206   
35082                          NaN           False      97616  45.118550   
35083                          NaN           False      97617  45.134428   

        latitude  superficie_km2  densite    nom_elu type_elu  
115    46.153426            16.0     53.0     BOULON    Maire  
116    46.009188             9.0     29.0      ORSET    Maire  
117    45.959530            24.0    607.0      FABRE    Maire  
118    45.996180            16.0    118.0     FORNES    Maire  
119    45.749499             6.0     19.0     BIONDA    Maire  
...          ...             ...      ...        ...      ...  
35079 -12.751310            22.0    298.0   IBRAHIMA    Maire  
35080 -12.837096            18.0    558.0      AMBDI    Maire  
35081 -12.796135             4.0   2686.0  MADI SOUF    Maire  
35082 -12.861165            11.0   1028.0   ABDALLAH    Maire  
35083 -12.782167            34.0    407.0  ISSILAMOU    Maire  

[34969 rows x 18 columns]